lib/streamlit/state/session_state.py

```diff
@@ -80,57 +83,75 @@ class Value:
 
 @attr.s(auto_attribs=True, slots=True, frozen=True)
 class WidgetMetadata:
+    """Metadata associated with a single widget. Immutable."""
+
     id: str
     deserializer: WidgetDeserializer = attr.ib(repr=False)
     serializer: WidgetSerializer = attr.ib(repr=False)
     value_type: Any
 
+    # An optional user-code callback invoked when the widget's value changes.
+    # Widget callbacks are called at the start of a script run, before the
+    # body of the script is executed.
     callback: Optional[WidgetCallback] = None
     callback_args: Optional[WidgetArgs] = None
     callback_kwargs: Optional[WidgetKwargs] = None
 
 
 @attr.s(auto_attribs=True, slots=True)
 class WStates(MutableMapping[str, Any]):
+    """A mapping of widget IDs to values. Widget values can be stored in
+    serialized or deserialized form, but when values are retrieved from the
+    mapping, they'll always be deserialized.
+    """
+
     states: Dict[str, WState] = attr.Factory(dict)
     widget_metadata: Dict[str, WidgetMetadata] = attr.Factory(dict)
 
     def __getitem__(self, k: str) -> Any:
-        item = self.states.get(k)
-        if item is not None:
```

Comment:
I reorganized this function to remove the deeply nested if/else clauses because I found it easier to reason about.